### PR TITLE
Silence gcc 12 and 13 warnings

### PR DESCRIPTION
**Pull request by wthrowe**
At least on my machine, there is still a runtime test failure as described in #5145.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Options/Auto.hpp [hunk 1]
```diff
@@ -43,6 +43,18 @@ class Auto {
   Auto() = default;
   explicit Auto(T value) : value_(std::move(value)) {}
 
+  // These lines are just to work around a spurious warning.
+  Auto(Auto&&) = default;
+  Auto& operator=(Auto&&) = default;
+#if defined(__GNUC__) and not defined(__clang__) and __GNUC__ >= 12 and __GNUC__ < 13
```
Comment (knelli2) [inline src/Options/Auto.hpp]: Isn't this just `__GNUC__ == 12`? (if so, same for the rest of this PR)

Comment (wthrowe) [reply src/Options/Auto.hpp]: It is.  I often write these things as `>= first_broken and < first_optimistically_fixed` (optimistically assuming untested versions have been fixed).  In this case with only one version it's kind of silly, so I'll change it if you like.

Comment (knelli2) [reply src/Options/Auto.hpp]: If you could, yeah so it's obvious it's just this one GCC version. If in the future, we find this is an issue with GCC 13 we can go back to this.

**Comment by wthrowe** [comment]
Changed the conditional formatting, and decided while I was at this I'd do gcc 13 too (12.2.1 and 13.1.1, specifically).  Version 13 actually caught some minor bugs.

**Review by knelli2 — APPROVED** [review]
Looks good. Not sure what the python formatting errors are about. Is this rebased on latest develop?

**Comment by kidder** [comment]
Ignoring python formatting issue (#5170)

**Pull request merged**